drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -229,7 +230,11 @@ static void nrf5_get_capabilities_at_boot(void)
 		((caps & NRF_802154_CAPABILITY_DELAYED_TX) ? IEEE802154_HW_TXTIME : 0UL) |
 		((caps & NRF_802154_CAPABILITY_DELAYED_RX) ? IEEE802154_HW_RXTIME : 0UL) |
 		IEEE802154_HW_SLEEP_TO_TX |
-		((caps & NRF_802154_CAPABILITY_SECURITY) ? IEEE802154_HW_TX_SEC : 0UL);
+		((caps & NRF_802154_CAPABILITY_SECURITY) ? IEEE802154_HW_TX_SEC : 0UL)
+#if defined(CONFIG_IEEE802154_NRF5_MULTIPLE_CCA)
```

Comment:
I'm wondering if we need this symbol at all. The nRF driver can support it unconditionally, the memory & processing overheads seem negligible. What do you think?

Reply:
I think that as this is not a standard, specification compliant feature it should be optional.

Reply:
The nRF driver needs to be deployed first, won't compile otherwise.

Reply:
I think that as this is not a standard, specification compliant feature it should be optional.

Reply:
> The nRF driver needs to be deployed first, won't compile otherwise.

True, however it's a matter of several days. If that's the only reason for introducing this switch then in my opinion we can wait and remove it.

> I think that as this is not a standard, specification compliant feature it should be optional.

It's already supported unconditionally by the driver, the capability is controlled by ieee802154 layer. What's the benefit from the possibility to disable it here?

Reply:
If you don't use this feature you can keep turned it off, this saves ram&flash. Linker won't remove this code because implementation is referenced through function pointers. 
Even if you never call 
`radio_api->configure` with `IEEE802154_EXT_OT_CONFIG_MULTIPLE_CCA` or 
`radio_api->tx` with mode == IEEE802154_EXT_OT_TX_MODE_TXTIME_MULTIPLE_CCA
the code handling this cases will need to remain there.

Reply:
I agree with you. My point was that these savings are negligible and not worth the additional configuration, especially that the driver below supports the feature unconditionally anyway, so disabling the option here doesn't lead to any more savings there.

I'm not blocking the PR over it ofc.